src/_P005_DHT.ino

```diff
@@ -149,28 +149,32 @@ bool P005_do_plugin_read(struct EventStruct *event) {
 
   pinMode(Plugin_005_DHT_Pin, OUTPUT);
   digitalWrite(Plugin_005_DHT_Pin, LOW);              // Pull low
+  
   switch (Par3) {
-    case P005_DHT11:
-    case P005_DHT22:
-    case P005_DHT12:  delay(18); break;  // FIXME TD-er: Must this be so long?
+    case P005_DHT11:  delay(18); break;  // minimum 18ms
+    case P005_DHT22:  delay(2);  break;  // minimum 1ms
+    case P005_DHT12:  delay(200); break; // minimum 200ms
     case P005_AM2301: delayMicroseconds(900); break;
     case P005_SI7021: delayMicroseconds(500); break;
   }
+  
+  digitalWrite(Plugin_005_DHT_Pin, HIGH);
+  
   switch (Par3) {
     case P005_DHT11:
     case P005_DHT22:
     case P005_DHT12:
     case P005_AM2301:
-      pinMode(Plugin_005_DHT_Pin, INPUT);
       delayMicroseconds(50);
       break;
     case P005_SI7021:
       // See: https://github.com/letscontrolit/ESPEasy/issues/1798
-      digitalWrite(Plugin_005_DHT_Pin, HIGH);
       delayMicroseconds(20);
-      pinMode(Plugin_005_DHT_Pin, INPUT);
       break;
   }
+  
+  pinMode(Plugin_005_DHT_Pin, INPUT);
```

Comment:
With this you are changing behavior of the AM2301.
That one was working fine, right?
Even though it does make sense to have it more symmetric.

Reply:
It really does not change anything as there is almost no difference in order when you delay then go into input or go into input and delay.
Datasheet says after initial request host must change pin to HIGH state.

Reply:
@TD-er if you are able to please create testbuild for @TheAppService

Reply:
Test build in progress....